components/maincontroller.js

```diff
@@ -644,6 +644,21 @@
 
         profile.TranscodingProfiles = profile.TranscodingProfiles.filter(item => item.Container !== "mkv");
 
+        //Another temporary solution until we can make a custom browserdeviceprofile
+        //1st and 2nd gen only support h264 4.1 but browserdeviceprofile is reporting 4.2
+        let context = cast.framework.CastReceiverContext.getInstance();
+        if (!context.canDisplayType('audio/mp4;codecs=avc1.64002A')) {
+            for (prof of profile.CodecProfiles) {
+                if (prof.Codec && prof.Codec.indexOf("h264") != -1) {
+                    for (condition of prof.Conditions) {
+                        if (condition.Property == "VideoLevel") {
+                            condition.Value = (condition.Value > 41)? 41 : condition.Value;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
I would write it like this I think
```suggestion
        if (context.canDisplayType('audio/mp4;codecs=avc1.64002A')) {
            return profile;
        }
        
        for (prof of profile.CodecProfiles) {
                if (prof.Codec && prof.Codec.indexOf("h264") == -1) {
                    continue;
                }
                for (condition of prof.Conditions) {
                    if (condition.Property == "VideoLevel" && condition.Value > 41) {
                        condition.Value = 41;
                    }
                }
            }
        }
```

Reply:
That's much cleaner, I'll go with that 👍